### DISCUSSION TITLE
Zigbee2Tasmota compatibility with Z-Stack 3 (Z-Stack 3.0.x and Z-Stack 3.x.x), i.e. Texas Instruments Zigbee 3.0 stack?

### DISCUSSION THREAD

**Discussion by Hedda** [arendst/Tasmota#12457] [Ideas]
Noticed merged https://github.com/arendst/Tasmota/pull/11016 so got to wondering if tasmota-zbbridge is now considered compatible with Z-Stack 3 (Z-Stack 3.x / Z-Stack 3.x.x)?

Z2T documentation does currently not mention TI's CC26x2 or CC13x2 hardware as of yet -> https://tasmota.github.io/docs/Zigbee/

There is also no [Z-Stack 3.x.0 coordinator firmware](https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator) for CC26x2 or CC13x2 -> https://github.com/arendst/Tasmota/tree/development/tools

That is Texas Instruments Zigbee 3.0 stack that CC2652/CC2652P/CC2652R/CC2652RB and CC1352/CC1352P MCUs uses by default.

Z2T documentation does not mention that it was only compatible or stable with the Z-Stack Home 1.2 FW and not Z-Stack 3 FW?

To make a simple and relatively inexpensive DIY tasmota-zbbridge (or tasmota32-zbbridge) based Zigbee to bridge with an Ethernet using a CC2652P RF module you can today buy cheap SMD Zigbee radio modules like "RFSTAR RF-BM-2652P2", "Ebyte E72-2G4M20S1E", or "RFSTAR RF-TI1352P1" and use one of those in combination with a ESP32 development board like example Wireless-Tag WT32-ETH01, Olimex ESP32-POE-ISO, LILYGO TTGO T-Internet-POE, (which are the type of modules that projects like Tube's Zigbee Gateways open source hardware are based on https://github.com/tube0013/tube_gateways and as such are components that have proven to make for a stable result, and as I understand, Silabs EFR32 variant have already been added to Tasmota compatibility list as per https://github.com/tube0013/tube_gateways/issues/1 ).

https://www.szrfstar.com/product/237-en.html

https://www.ebyte.com/en/product-view-news.html?id=1002

https://www.rfstariot.com/product/235.html

Note! While not recommended for a "production" Zigbee network it is possible for testing and development purposes to flash CC2538, CC2531 and CC2530 based modules and adapters with Z-Stack 3.0.x firmware, (if you today have a CC253x module/adapter that is already flashed with Z-Stack Home 1.2 firmware then you should be able to upgrade to a Z-Stack 3.0.x firmware, but note that such upgrade will, however, probably reset your NVRAM and Zigbee network so backup that before).

https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator

https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin

By the way, looks like Texas Instruments have actually deprecated Z-Stack Home 1.2? -> https://www.ti.com/tool/Z-STACK-ARCHIVE

FYI, Texas Instruments modules/adapters based on CC2652P/CC2652R/CC2652RB are growing in popularity as Zigbee 3.0 compliant Zigbee coordinators among other open source home automation projects/communities (inc. Zigbee2MQTT and Home Assistant).

PS: Zigbee2MQTT devs have lists of TI adapters made for Z-Stack 3 (Z-Stack 3.x.x) as well as precompiled coordinator firmware images:

https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin

https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/bin/README.md

Adapter | TI Chip/Module Used | Firmware to Flash | BSL Trigger Pin (1) | Auto-BSL (2) | RF Switch Control Pins (3) | LED(s)
-- | -- | -- | -- | -- | -- | --
TI LAUNCHXL-CC26xR1 | CC2652R | CC2652R_*.zip | DIO_13 | No | N/A | DIO_6 (Red)DIO_7 (Green)
TI LAUNCHXL-CC1352P-2 | CC1352P | CC1352P2_CC2652P_launchpad_*.zip | DIO_15 | No | DIO_28: 2.4GhzDIO_29: 20dBm PADIO_30: Sub-1GHz | DIO_6 (Red)DIO_7 (Green)
Electrolama zzh | CC2652R | CC2652R_*.zip | DIO_13 | No | N/A | DIO_7 (Pink)
Electrolama zzhp-lite | CC2652P(Ebyte E72) | CC1352P2_CC2652P_other_*.zip | DIO_15 | Yes | DIO_5: 20dBm PA ??DIO_6: 2.4GHz ?? | DIO_7 (Pink)
Electrolama zzhp | CC2652P | CC1352P2_CC2652P_other_*.zip | DIO_15 | Yes | DIO_5: 20dBm PA ??DIO_6: 2.4GHz ?? | DIO_7 (Pink)
Electrolama zoe2 | CC1352P(Ebyte E79) | CC1352P2_CC2652P_other_*.zip | DIO_15 | No | DIO_5: 20dBm PA ??DIO_6: 2.4GHz ?? | DIO_7 (Pink)
Slaesh's CC2652RB stick | CC2652RB | CC2652RB_*.zip | DIO_13 | Yes | N/A | DIO_7 (Blue)
ZigStar Stick v4 | CC2652P(RFSTAR RF-BM-2652P2) | CC1352P2_CC2652P_launchpad_*.zip | DIO_15 | Only for CH340C ver. | DIO_28: 2.4GhzDIO_29: 20dBm PA | DIO_6 (Green)DIO_7 (Red)
CircuitSetup.us Zigbee Stick | CC2652P (RFSTAR RF-BM-2652P2 PCB Ant.) | CC1352P2_CC2652P_launchpad_*.zip | DIO_15 | Only for CH340C ver. | DIO_28: 2.4GhzDIO_29: 20dBm PA | DIO_6 (Green)DIO_7 (Red)
Tube's CC2652P2 USB Coordinator | CC2652P(RFSTAR RF-BM-2652P2) | CC1352P2_CC2652P_launchpad_*.zip | DIO_15 | N/A | DIO_28: 2.4GhzDIO_29: 20dBm PA | N/A
Tube's Zigbee Gateways (CC2652P2 variant) | CC2652P(RFSTAR RF-BM-2652P2) | CC1352P2_CC2652P_launchpad_*.zip | DIO_15 | N/A | DIO_28: 2.4GhzDIO_29: 20dBm PA | N/A
Tube's Zigbee PoE (Power Over Ethernet) Serial Coordinator (CC2652P2 variant) | CC2652P(RFSTAR RF-BM-2652P2) | CC1352P2_CC2652P_launchpad_*.zip | DIO_15 | N/A | DIO_28: 2.4GhzDIO_29: 20dBm PA | N/A
Egony Stick V4 (Ebyte ver.) | CC2652P(Ebyte E72-2G4M20S1E) | CC1352P2_CC2652P_other_*.zip | DIO_15 | Yes(from Rev.2.0) | DIO_5: 20dBm PADIO_6: 2.4GHz | DIO_8 (Green)DIO_7 (Red)
Gio-dot Z-Bee Duo with CC2652P | CC2652P(Ebyte E72-2G4M20S1E) | CC1352P2_CC2652P_other_*.zip | DIO_15 | Yes(from Rev.2.0) | DIO_5: 20dBm PADIO_6: 2.4GHz | DIO_8 (Green)DIO_7 (Red)
Egony Stick V4 (RFSTAR ver.) | CC2652P(RFSTAR RF-BM-2652P2) | CC1352P2_CC2652P_launchpad_*.zip | DIO_15 | Yes | DIO_28: 2.4GhzDIO_29: 20dBm PA | DIO_7 (Green)DIO_6 (Red)
cod.m Zigbee CC2652P RPi Module| CC2652P(RFSTAR RF-BM-2652P2) | CC1352P2_CC2652P_launchpad_*.zip | DIO_15 | Yes | DIO_28: 2.4GhzDIO_29: 20dBm PA | DIO_7 (Green)DIO_6 (Red)
Gio-dot Z-Bee Duo with CC2652P| CC2652P (Ebyte E72-2G4M20S1E) | CC1352P2_CC2652P_other_*.zip | DIO_15 | Yes (from Rev.2.0) | DIO_5: 20dBm PA | DIO_8 (Green) DIO_7 (Red)
cyijun OpenZ3Gateway| CC2652P (RFSTAR RF-BM-2652P2 SMA Ant.) | CC1352P2_CC2652P_launchpad_*.zip | DIO_15 | No | DIO_28: 2.4Ghz DIO_29: 20dBm PA | DIO_7 (Green) DIO_6 (Red)
SONOFF Zigbee 3.0 USB Dongle Plus| CC2652P | CC1352P2_CC2652P_launchpad_*.zip | ? | No | ? | ?

More information on those modules and adapters here -> https://www.zigbee2mqtt.io/information/supported_adapters.html

**Comment by digiblur** [comment]
Even iTead is jumping on the CC2652 chipset series train.

**Reply by Hedda** [reply]
Yes I head that ITead (of Sonoff fame) are currently in the process of making a TI CC2652P based USB dongle, which presumably will have serial TX and RX solder-pads as well so could optionally be used to build your own DIY Z2T bridge with an ESP32 or ESP8266. At least their current [Silicon Labs EFR32MG21 Zigbee 3.0 USB Dongle board](https://itead.cc/product/zigbee-3-0-usb-dongle/) features such serial TX and RX solder-pads on its back.

Off-topic; By the way, I also heard that ITead is currently also in the process of making a Silabs EFR32MG21 "Plus" USB dongle too, and ITead is also using the same Silabs EFR32MG21 chip in other devices so I doubt they would stop continuing making those now.

I think ITead is just broadening its use of different Zigbee chips to include other manufactures as it is well known that Silicon Labs suffered more than most in the ongoing 2020–2021 global chip shortage, however, I bet that if the devices sell well then ITead will continue to make both Silabs and TI based devices once the Silabs chip shortage is not so dire.

**Reply by digiblur** [reply]
Definitely do not want all the eggs in one basket.

**Comment by Jason2866** [comment]
CC2652 needs to be flashed with Koenkk Z-Stack firmware to be used with z2t.
The CC2652 is in general supported from z2t but not tested. So it is not in the Tasmota documentation.
Testing is welcome and if everything is working as expected contributing to the docu how all is done is welcome.

**Reply by Hedda** [reply]
> CC2652 needs to be flashed with Koenkk Z-Stack firmware to be used with z2t.

FYI, those mentioned do all come pre-flashed with Koenkk Z-Stack 3.x.0 firmware by default (and ITead's new dongle will too):

https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/bin/README.md

https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin

https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/README.md

**Reply by Jason2866** [reply]
Great, so just use / test it with Tasmota

**Comment by Hedda** [comment]
FYI, s-hadinger confirm Zigbee2Tasmota is compatible with ZStack 3 and works with CC2652

https://github.com/tasmota/docs/pull/688
